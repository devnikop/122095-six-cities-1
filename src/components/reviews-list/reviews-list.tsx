import { connect } from 'react-redux';
import * as React from 'react';

import { Comment } from '../../types';

import { getReviews } from '../../reducer/data/selectors';
import { Operation } from '../../reducer/data/data';

import ReviewItem from '../reviews-item/reviews-item';

interface Props {
  offerId: number,
  onLoadReviews: (id: number) => void,
  reviews: Comment[],
}

class ReviewList extends React.PureComponent<Props> {
  render() {
    const {
      reviews,
    } = this.props;

    return <>
      <h2 className="reviews__title">Reviews &middot;
      <span className="reviews__amount">{reviews.length}</span>
      </h2>
      <ul className="reviews__list">
        {reviews.map((review, it) => <ReviewItem
          key={`review${it}`}
          review={review}
        />)}
      </ul>
    </>;
  }

  componentDidMount() {
    const {
      offerId,
      onLoadReviews,
    } = this.props;

    onLoadReviews(offerId);
  }
}

const mapStateToProps = (state) => ({
  reviews: getReviews(state),
});

const mapDispatchToProps = (dispatch) => ({
  onLoadReviews: (id) => dispatch(Operation.loadReviews(id)),
});

export { ReviewList };

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(ReviewList);
