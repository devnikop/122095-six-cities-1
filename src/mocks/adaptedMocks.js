const adaptedLoginDataMock = {
  avatarUrl: `/static/avatar/5.jpg`,
  email: `doppervily@yandex.ru`,
  id: 1,
  isPro: false,
  name: `doppervily`,
};

const adaptedOfferMock = {
  bedrooms: 1,
  city: {
    coords: [48.85661, 2.351499],
    zoom: 13,
    name: `Paris`,
  },
  description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
  goods: [`Towels`, `Laptop friendly workspace`, `Baby seat`, `Washer`, `Breakfast`, `Air conditioning`],
  host: {
    avatar: `img/avatar-angelina.jpg`,
    id: 25,
    name: `Angelina`,
    isPro: true,
  },
  id: 1,
  images: [`https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`],
  isFavorite: true,
  isPremium: false,
  maxAdults: 3,
  place: {
    coords: [48.834610000000005, 2.364499],
    zoom: 16,
  },
  previewImage: `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`,
  price: 231,
  rating: 5,
  title: `Canal View Prinsengracht`,
  type: `room`,
};

const adaptedOffersArrayMock = [
  {
    bedrooms: 1,
    city: {
      coords: [48.85661, 2.351499],
      zoom: 13,
      name: `Paris`,
    },
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    goods: [`Towels`, `Laptop friendly workspace`, `Baby seat`, `Washer`, `Breakfast`, `Air conditioning`],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      id: 25,
      name: `Angelina`,
      isPro: true,
    },
    id: 1,
    images: [`https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`],
    isFavorite: true,
    isPremium: false,
    maxAdults: 3,
    place: {
      coords: [48.834610000000005, 2.364499],
      zoom: 16,
    },
    previewImage: `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`,
    price: 231,
    rating: 5,
    title: `Canal View Prinsengracht`,
    type: `room`,
  },
];

const adaptedReviewsArrayMock = [
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`,
    id: 1,
    rating: 4,
    user: {
      avatarUrl: `img/1.png`,
      id: 4,
      isPro: false,
      name: `Max`,
    },
  },
];

export {
  adaptedLoginDataMock,
  adaptedOfferMock,
  adaptedOffersArrayMock,
  adaptedReviewsArrayMock,
};
