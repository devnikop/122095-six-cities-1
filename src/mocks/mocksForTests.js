const leafletMock = {
  icon() {
    return {
      iconSize: [30, 30],
      iconUrl: `img/pin.svg`,
    };
  },
  map() {
    return {
      setView: jest.fn(),
    };
  },
  tileLayer() {
    return {
      addTo: jest.fn(),
    };
  },
  layerGroup() {
    return {
      addTo: jest.fn(),
    };
  },
  marker() {
    return {
      addTo: jest.fn(),
    };
  },
};

const offersArrayMock = [
  {
    city: `Amsterdam`,
    coords: [52.3909553943508, 4.85309666406198],
    isPremium: true,
    placeName: `Beautiful & luxurious apartment at great location`,
    placeType: `Apartment`,
    price: 120,
    src: `img/apartment-01.jpg`,
  },
  {
    city: `Cologne`,
    coords: [52.369553943508, 4.85309666406198],
    isPremium: false,
    placeName: `Wood and stone place`,
    placeType: `Private room`,
    price: 80,
    src: `img/room.jpg`,
  },
  {
    city: `Brussels`,
    coords: [52.3909553943508, 4.929309666406198],
    isPremium: false,
    placeName: `Canal View Prinsengracht`,
    placeType: `Apartment`,
    price: 132,
    src: `img/apartment-02.jpg`,
  },
  {
    city: `Amsterdam`,
    coords: [52.3809553943508, 4.939309666406198],
    isPremium: true,
    placeName: `Nice, cozy, warm big bed apartment`,
    placeType: `Apartment`,
    price: 180,
    src: `img/apartment-03.jpg`,
  }
];

const citiesMock = [
  {
    coords: [52.370216, 4.895168],
    name: `Amsterdam`,
  },
  {
    coords: [50.850346, 4.351721],
    name: `Brussels`,
  },
  {
    coords: [51.227741, 6.773456],
    name: `Cologne`,
  },
  {
    coords: [50.937531, 6.960279],
    name: `Dusseldorf`,
  },
  {
    coords: [53.551086, 9.993682],
    name: `Hamburg`,
  },
  {
    coords: [48.856613, 2.352222],
    name: `Paris`,
  },
];

const offerMock = {
  isPremium: true,
  placeName: `Beautiful & luxurious apartment at great location`,
  placeType: `Apartment`,
  price: 120,
  src: `img/apartment-01.jpg`,
};

export {
  citiesMock,
  leafletMock,
  offerMock,
  offersArrayMock,
};
