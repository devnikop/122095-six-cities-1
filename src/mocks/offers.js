// const offers = [
//   {
//     city: `Amsterdam`,
//     placeName: `Beautiful & luxurious apartment at great location`,
//     placeType: `Apartment`,
//     isPremium: true,
//     src: `img/apartment-01.jpg`,
//     price: 120,
//     coords: [52.3909553943508, 4.85309666406198],
//   },
//   {
//     city: `Cologne`,
//     placeName: `Wood and stone place`,
//     placeType: `Private room`,
//     isPremium: false,
//     src: `img/room.jpg`,
//     price: 80,
//     coords: [52.369553943508, 4.85309666406198],
//   },
//   {
//     city: `Dusseldorf`,
//     placeName: `Beautiful & luxurious apartment at great location`,
//     placeType: `Apartment`,
//     isPremium: true,
//     src: `img/apartment-01.jpg`,
//     price: 120,
//     coords: [52.3909553943508, 4.85309666406198],
//   },
//   {
//     city: `Hamburg`,
//     placeName: `Wood and stone place`,
//     placeType: `Private room`,
//     isPremium: false,
//     src: `img/room.jpg`,
//     price: 80,
//     coords: [52.369553943508, 4.85309666406198],
//   },
//   {
//     city: `Amsterdam`,
//     placeName: `Beautiful & luxurious apartment at great location`,
//     placeType: `Apartment`,
//     isPremium: true,
//     src: `img/apartment-01.jpg`,
//     price: 120,
//     coords: [52.3909553943508, 4.85309666406198],
//   },
//   {
//     city: `Cologne`,
//     placeName: `Wood and stone place`,
//     placeType: `Private room`,
//     isPremium: false,
//     src: `img/room.jpg`,
//     price: 80,
//     coords: [52.369553943508, 4.85309666406198],
//   },
//   {
//     city: `Brussels`,
//     placeName: `Canal View Prinsengracht`,
//     placeType: `Apartment`,
//     isPremium: false,
//     src: `img/apartment-02.jpg`,
//     price: 132,
//     coords: [52.3909553943508, 4.929309666406198],
//   },
//   {
//     city: `Amsterdam`,
//     placeName: `Nice, cozy, warm big bed apartment`,
//     placeType: `Apartment`,
//     isPremium: true,
//     src: `img/apartment-03.jpg`,
//     price: 180,
//     coords: [52.3809553943508, 4.939309666406198],
//   }
// ];

// const cities = [
//   {
//     name: `Amsterdam`,
//     coords: [52.370216, 4.895168],
//   },
//   {
//     name: `Brussels`,
//     coords: [50.850346, 4.351721],
//   },
//   {
//     name: `Cologne`,
//     coords: [51.227741, 6.773456],
//   },
//   {
//     name: `Dusseldorf`,
//     coords: [50.937531, 6.960279],
//   },
//   {
//     name: `Hamburg`,
//     coords: [53.551086, 9.993682],
//   },
//   {
//     name: `Paris`,
//     coords: [48.856613, 2.352222],
//   },
//   {
//     name: `Moscow`,
//     coords: [48.856613, 2.352222],
//   },
// ];

// export {
//   offers,
//   cities
// };
