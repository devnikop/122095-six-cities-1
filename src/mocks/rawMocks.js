const rawLoginDataMock = {
  "avatar_url": `/static/avatar/5.jpg`,
  "email": `doppervily@yandex.ru`,
  "id": 1,
  "is_pro": false,
  "name": `doppervily`,
};

const rawOfferMock = {
  "bedrooms": 1,
  "city": {name: `Paris`, location: {latitude: 48.85661, longitude: 2.351499, zoom: 13}},
  "description": `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
  "goods": [`Towels`, `Laptop friendly workspace`, `Baby seat`, `Washer`, `Breakfast`, `Air conditioning`],
  "host": {"id": 25, "name": `Angelina`, "is_pro": true, "avatar_url": `img/avatar-angelina.jpg`},
  "id": 1,
  "images": [`https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`],
  "is_favorite": true,
  "is_premium": false,
  "location": {latitude: 48.834610000000005, longitude: 2.364499, zoom: 16},
  "max_adults": 3,
  "preview_image": `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`,
  "price": 231,
  "rating": 4.8,
  "title": `Canal View Prinsengracht`,
  "type": `room`
};

const rawOffersArrayMock = [
  {
    "bedrooms": 1,
    "city": {name: `Paris`, location: {latitude: 48.85661, longitude: 2.351499, zoom: 13}},
    "description": `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    "goods": [`Towels`, `Laptop friendly workspace`, `Baby seat`, `Washer`, `Breakfast`, `Air conditioning`],
    "host": {"id": 25, "name": `Angelina`, "is_pro": true, "avatar_url": `img/avatar-angelina.jpg`},
    "id": 1,
    "images": [`https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`],
    "is_favorite": true,
    "is_premium": false,
    "location": {latitude: 48.834610000000005, longitude: 2.364499, zoom: 16},
    "max_adults": 3,
    "preview_image": `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`,
    "price": 231,
    "rating": 4.8,
    "title": `Canal View Prinsengracht`,
    "type": `room`
  }
];

const rawReviewsArrayMock = [
  {
    "comment": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    "date": `2019-05-08T14:13:56.569Z`,
    "id": 1,
    "rating": 4,
    "user": {
      "avatar_url": `img/1.png`,
      "id": 4,
      "is_pro": false,
      "name": `Max`,
    },
  },
];

export {
  rawLoginDataMock,
  rawOfferMock,
  rawOffersArrayMock,
  rawReviewsArrayMock,
};
