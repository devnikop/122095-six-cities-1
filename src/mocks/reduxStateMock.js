import {
  citiesMock,
  offersArrayMock,
  reviewsArrayMock,
  loginDataMock,
} from './mocksForTests';

import NameSpace from '../reducer/name-spaces';

const StoreMock = {
  data: {
    activeOfferId: 0,
    currentCity: citiesMock[0],
    offers: offersArrayMock,
    sortedOffers: offersArrayMock,
  },
  favorite: {
    favoriteOffers: offersArrayMock,
  },
  review: {
    reviews: reviewsArrayMock,
  },
  user: {
    isAuthorizationRequired: false,
    loginData: loginDataMock,
  },
};

const reduxStateMock = {
  [NameSpace.DATA]: StoreMock.data,
  [NameSpace.FAVORITE]: StoreMock.favorite,
  [NameSpace.REVIEW]: StoreMock.review,
  [NameSpace.USER]: StoreMock.user,
};

export default reduxStateMock;
