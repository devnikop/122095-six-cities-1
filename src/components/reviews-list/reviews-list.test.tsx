import * as React from 'react';
import * as renderer from 'react-test-renderer';

import {ReviewList} from './reviews-list';

import {reviewsArrayMock} from '../../mocks/mocksForTests';

it(`ReviewList renders correctly`, () => {
  const tree = renderer
    .create(<ReviewList
      offerId={1}
      reviews={reviewsArrayMock}
      onLoadReviews={jest.fn()}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
