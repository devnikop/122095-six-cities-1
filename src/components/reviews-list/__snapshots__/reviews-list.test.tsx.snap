// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewList renders correctly 1`] = `
Array [
  <h2
    className="reviews__title"
  >
    Reviews Â·
    <span
      className="reviews__amount"
    >
      3
    </span>
  </h2>,
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/1.png"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Max
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "80%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
        </p>
        <time
          className="reviews__time"
          dateTime="2019-05-08T14:13:56.569Z"
        >
          April 2019
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/4.png"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Maenx
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "80%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Aijw3ifniknfii2222 a river by the unique lightness of Amsterdam.
        </p>
        <time
          className="reviews__time"
          dateTime="2019-05-08T14:13:56.569Z"
        >
          April 2019
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/9.png"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Mdgnx
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "20%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Asdaf f234frefdsafdsfggdam.
        </p>
        <time
          className="reviews__time"
          dateTime="2019-05-08T14:13:56.569Z"
        >
          April 2019
        </time>
      </div>
    </li>
  </ul>,
]
`;
